### PR TITLE
[WIP] Rivtech factory

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
# Summary:

Resurrection of #7953 (shout-out to @Frost-wood).

# What was changed:

- added material (`RUBBER`);
- added spare parts (plc, pulley, rubber belt);
- added item (broken proximity sensor);
- added terrains (conveyor and some machinery);
- added furniture (plc, ammuntion loader, hopper);
- added new monster attack (`PROXIMITY_ALARM`);
- added monsters (proximity alarm sensor and two type of armored zombies);
- added overmap special (Rivtech factory).

# Screenshots:

_No screenshots yet - I need to cleanup things first_

### CODE REVIEW

**Comment by smolbird** [comment]
Could change one of the repair items to allow repairing rubber items, most likely soldering iron since most existing "rubber" items are repaired with that.

Speaking of, we do have rubber items that are merely plastic in the JSON.

**Comment by DracoGriffin** [comment]
Great on picking up on this work. My only qualm is with the new rubber material. It should be noted somewhere that a lot of the current plastic material items may be needed to update to rubber which may drastically alter their stats unintendedly.

**Comment by BorkBorkGoesTheCode** [comment]
@ZhilkinSerg try linting the mapgen palette.

File: data/json/terrain.json [hunk 1]
```diff
@@ -6837,5 +6837,225 @@
...
+        { "item": "chain", "amount": 1, "chance": 60 },
+        { "item": "cu_pipe", "amount": 1, "chance": 20 },
+        { "item": "steel_lump", "amount": 2, "minamount": 1 },
+        { "item": "hose", "amount": 1 },
+        { "item": "sheet_metal", "amount": 3, "minamount": 1 },
+        { "item": "steel_chunk", "amount": 3, "minamount": 1 },
+        { "item": "bearing", "amount": 12, "minamount": 4 },
+        { "item": "frame", "amount": 1, "chance": 60 },
+        { "item": "motor", "amount": 1, "chance": 30 },
+        { "item": "metal_tank_small", "amount": 1, "chance": 30 },
```
Comment (DracoGriffin) [inline data/json/terrain.json]: Apparently this item is an issue for Travis:
`src/item_group.cpp:116 [virtual void Single_item_creator::check_consistency() const] item id metal_tank_small is unknown`

`src/item_group.cpp:116 [virtual void Single_item_creator::check_consistency() const] item id metal_tank_small is unknown`

-- However, it's listed in migration.json (Line 333):
```json
  {
    "id": "metal_tank_small",
    "type": "MIGRATION",
    "replace": "jerrycan"
  },
```

Replacing with jerrycan for the simple fix, but it shouldn't be throwing an error, right?

**Comment by ZhilkinSerg** [comment]
I will put it back in a queue for a while as it needs a lot of overmap terrains in addition to existing ones.

**Comment by Throwaway-name** [comment]
@ZhilkinSerg

**Pull request closed without merging**